[libc++][lldb-dap][test] Fix CI for bootstrapping-build

frederick-vs-ja opened the pull request:
#140107 changed the default argument of `disableASLR` of related functions from `False` to `True`. libc++ CI has been stably failing for `TestDAP_subtleFrames.py` (in bootstrapping-build) since then. The error message "personality set failed: Operation not permitted" seems related to ASLR.

This PR attempts to fix the CI failure by changing the default value of `disableASLR` in `dap_server.py` to `False`.

frederick-vs-ja committed:
[libc++][lldb-dap][test] Fix CI for bootstrapping-build

frederick-vs-ja commented:
There're many TSan and MSan errors seemingly related to ASLR. But it's curious that how did they happen when `TestDAP_subtleFrames.py` hasn't been executed.

Michael137 commented:
I'm a bit confused why this test is special and would be the only one manually setting `disableASLR=False`. I changed the default in https://github.com/llvm/llvm-project/pull/113593

@JDevlieghere @ashgti why did we switch the default back to `disableASLR=False`?

Michael137 commented:
Looks like https://github.com/llvm/llvm-project/pull/140107 refactored this stuff and as @frederick-vs-ja points out, my fix in https://github.com/llvm/llvm-project/pull/113593 essentially got reverted. @ashgti is there some default config for tests that we could stick `disableASLR=False` into?

ashgti commented:
Maybe we should update https://github.com/llvm/llvm-project/blob/ff7bb17c88328276323603809d5d4549ca8bd22b/lldb/packages/Python/lldbsuite/test/tools/lldb-dap/dap_server.py#L859C9-L859C20 to default to `False` instead of `True`?

Michael137 commented:
> Maybe we should update https://github.com/llvm/llvm-project/blob/ff7bb17c88328276323603809d5d4549ca8bd22b/lldb/packages/Python/lldbsuite/test/tools/lldb-dap/dap_server.py#L859C9-L859C20 to default to `False` instead of `True`?

Yea i think that's the best way forward

frederick-vs-ja committed:
Merge branch 'main' into libcxx-lldb-bootstrapping-build-ci-fix

frederick-vs-ja committed:
Adopt @ashgti‘s review comments

- Change the default argument for `disableASLR` in `dap_server.py`.
- Revert change in `TestDAP_subtleFrames.py`.

frederick-vs-ja committed:
Merge branch 'main' into libcxx-lldb-bootstrapping-build-ci-fix

ldionne commented:
I'm going to land this to attempt to unbreak libc++'s CI, which is in a really bad shape right now. From the comments above, this seems to have consensus to move forward.

ldionne committed:
Remove comment
